Assets/Scripts/Core/SpacialGravityReceiver.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+
+/// <summary>
+/// Attach this to any GameObject you want to experience spacial
+/// gravity from a CircleCollider2D.  The gravity acts as a force
+/// applied to the RigidBody2D this component is a sibling of.  Note
+/// that the force experienced will scaled linearly with the mass of
+/// the rigidbody.
+/// </summary>
+[RequireComponent(typeof(Rigidbody2D))]
+[RequireComponent(typeof(Collider2D))]
+public class SpacialGravityReceiver : MonoBehaviour
+{
+    private Dictionary<int, CircleGravity2D> _activeGravityRegions;
+    private Rigidbody2D _rigidbody;
+
+    private Vector2 CurrentAcceleration()
+    {
+        Vector2 acceleration = Vector2.zero;
+        foreach (KeyValuePair<int, CircleGravity2D> gravityRegion
```

Comment:
Doing this as a `Dictionary.Reduce()` might be more clear than the for loop and accumulator.

Reply:
Doesn't look like C# has a nice way to do that